components/tfx/Evaluator/with_URI_IO/component.py

```diff
@@ -0,0 +1,103 @@
+# flake8: noqa
+
+from typing import NamedTuple
+
+def Evaluator(
+    examples_uri: 'ExamplesUri',
+    model_uri: 'ModelUri',
+    output_evaluation_uri: 'ModelEvaluationUri',
+    #output_blessing_uri: 'ModelBlessingUri',
+    baseline_model_uri: 'ModelUri' = None,
+    schema_uri: 'SchemaUri' = None,
+    eval_config: {'JsonObject': {'data_type': 'proto:tensorflow_model_analysis.EvalConfig'}} = None,
+    feature_slicing_spec: {'JsonObject': {'data_type': 'proto:tfx.components.evaluator.FeatureSlicingSpec'}} = None,
+    fairness_indicator_thresholds: list = None,
+    beam_pipeline_args: list = None,
+) -> NamedTuple('Outputs', [
+    ('evaluation_uri', 'ModelEvaluationUri'),
+    #('blessing_uri', 'ModelBlessingUri'),
+]):
+    from tfx.components import Evaluator as component_class
+
+    #Generated code
+    import json
+    import os
+    import tempfile
+    import tensorflow
+    import uuid
+    from google.protobuf import json_format, message
+    from tfx.types import channel_utils, artifact_utils
+    from tfx.components.base import base_executor
+
+    arguments = locals().copy()
+
+    component_class_args = {}
+
+    for name, execution_parameter in component_class.SPEC_CLASS.PARAMETERS.items():
+        argument_value_obj = argument_value = arguments.get(name, None)
+        if argument_value is None:
+            continue
+        parameter_type = execution_parameter.type
+        if isinstance(parameter_type, type) and issubclass(parameter_type, message.Message): # Maybe FIX: execution_parameter.type can also be a tuple
+            argument_value_obj = parameter_type()
+            json_format.Parse(argument_value, argument_value_obj)
+        component_class_args[name] = argument_value_obj
+
+    for name, channel_parameter in component_class.SPEC_CLASS.INPUTS.items():
+        artifact_path = arguments.get(name + '_uri', None) or arguments.get(name + '_path', None)
+        if artifact_path:
+            artifact = channel_parameter.type()
+            artifact.uri = artifact_path + '/' # ?
+            if channel_parameter.type.PROPERTIES and 'split_names' in channel_parameter.type.PROPERTIES:
+                # Recovering splits
+                subdirs = tensorflow.io.gfile.listdir(artifact_path)
+                # Workaround for https://github.com/tensorflow/tensorflow/issues/39167
+                subdirs = [subdir.rstrip('/') for subdir in subdirs]
+                artifact.split_names = artifact_utils.encode_split_names(sorted(subdirs))
+            component_class_args[name] = channel_utils.as_channel([artifact])
+
+    component_class_instance = component_class(**component_class_args)
+
+    input_dict = {name: channel.get() for name, channel in component_class_instance.inputs.get_all().items()}
```

Comment:
Does it help to use `tfx.types.channel_utils.unwrap_channel_dict` to make it more succinct?

Reply:
Thanks for reminding.
I'd like to.
The problem is that `component_class_instance.inputs` is not a dict >_<
It's a `_PropertyDictWrapper` which does not have `.items()` that `unwrap_channel_dict` uses.
And `component_class_instance._input_dict` is private.

I can try fixing this on the TFX side and then these components can switch to it.

Reply:
maybe `input_dict = channel_utils.unwrap_channel_dict(component_class_instance.inputs.get_all())`? You can do the same to `output_dict` as well.

Reply:
Done.
I wonder how stable that API is...